### PR TITLE
Convert issue templates to issue forms

### LINKED ISSUE #5470: Use forms instead of Markdown templates
## The problem with Markdown templates

This project still uses Markdown templates, like this:

```markdown
<!-- For support on the GitHub-integrated Dependabot service, please contact [GitHub support](https://support.github.com/) -->
<!-- This issue-tracker is meant for issues related to Dependabot's updating logic, a good rule of thumb is that if you have questions about the _diff_ in a PR, it belongs here, otherwise the GitHub support team is best equipped to help you -->

<!-- The more information you can provide, the easier it will be to reproduce the issue and find a fix -->

**Package ecosystem**
<!-- npm, docker, bundler, etc. -->
**Package manager version**
<!-- If applicable, specify the package manager version you're using (e.g., npm 7.1, pip-compile 5.0, etc.) -->
**Language version**
<!-- If applicable, specify the language version you're using (e.g., node 14.1, Ruby 2.7, etc. ) -->
**Manifest location and content before the Dependabot update**
<!-- If applicable, specify the path to each manifest file (/client/package.json, /Gemfile, etc.) -->
<!-- If applicable, attach each manifest file or provide a link to each manifest file -->
**dependabot.yml content**
<!-- If using GitHub-native Dependabot, attach your dependabot.yml file or provide a link to it -->
**Updated dependency**
<!-- If applicable, the dependency name and to and from versions -->
**What you expected to see, versus what you actually saw**
<!-- A clear and concise description of what you expected to happen -->
**Native package manager behavior**
<!-- If applicable, what output do you see when you update the dependency using the native package manager (e.g., bundler, npm, etc.)? -->
**Images of the diff or a link to the PR, issue, or logs**
<!-- If applicable, add links to public PRs or Issues that Dependabot opened, and/or paste in any related logs -->
**🕹 Bonus points: Smallest manifest that reproduces the issue**
<!-- Optional: Want to make a :robot: smile? You can do it! Show us the smallest manifest file (or set of files) that reproduces this problem. Remove anything that's not essential to reproduce this issue, and it will help us address the problem more quickly. ✨ -->
```

Problems:

- Hard to read the template
- Easy to accidentally drop wanted fields/descriptions
- Having a wall of text can turn people off from reporting a bug (it did for me at least)

For these reasons, I propose this project uses Issue Forms instead.

### Pros/cons of using forms

Pros:

- Better layout
- Dropdown with values
- Checkboxes
- Single fields
- Multiple lines of text fields
- Descriptions/background information above the field
- Make certain fields/checkboxes mandatory
- Users can't mangle forms as easily

Cons:

- Writing the YAML for the forms is a bit tricky
- Issue Forms are still in beta, officially

## Documentation

- [GitHub docs, syntax for issue forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms)
- [GitHub Docs, syntax for GitHub's form schema](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema)

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Convert issue templates to issue forms

## Context:

Closes #5470.

You can preview the forms by going to my fork [^fork], and clicking on the `.yml` files.

[^fork]: https://github.com/HonkingGoose/dependabot-core/tree/chore/use-forms/.github/ISSUE_TEMPLATE

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+name: \U0001F41B Bug report
```
Comment (deivid-rodriguez) [inline .github/ISSUE_TEMPLATE/bug-report.yml:1]: These seem to no longer render fine in forms? I guess we can copy the rendered emoji directly like with the ladybug below.

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+        The more information you can provide, the easier it will be to reproduce the issue and find a fix.
+
+  - type: checkboxes
+    attributes:
+      label: Is there an existing issue for this?
+      description: Please search existing issues to avoid creating duplicates.
+      options:
+        - label: I have searched the existing issues
+          required: true
```
Comment (deivid-rodriguez) [inline .github/ISSUE_TEMPLATE/bug-report.yml:19]: We were not mentioning this before in the previous bug template, thanks for adding it! Also makes sense to me to make it required.

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 3]
```diff
@@ -1,15 +0,0 @@
...
-title: ''
-labels: 'T: feature-request'
-assignees: ''
-
----
-
-<!-- Please search existing issues to avoid creating duplicates. -->
-
-<!-- The Dependabot team is currently at reduced capacity, because of this our
-response times on issues will be slower than we'd like. -->
```
Comment (deivid-rodriguez) [inline .github/ISSUE_TEMPLATE/feature-request.md:13]: Seems fine to remove the above note 👍.

**Comment by jurre** [comment]
Thanks for doing this, I think it's a great change! ❤️

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 2 above)
Comment (jurre) [reply .github/ISSUE_TEMPLATE/bug-report.yml:19]: Agreed

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 1 above)
Comment (HonkingGoose) [reply .github/ISSUE_TEMPLATE/bug-report.yml:1]:
```suggestion
name: 🐛 Bug report
```

File: .github/ISSUE_TEMPLATE/feature-request.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+name: \U0001F680 Feature request
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/feature-request.yml:1]:
```suggestion
name: 🚀 Feature request
```

File: .github/ISSUE_TEMPLATE/tech-debt.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+name: \U00002699 Tech Debt
```
Comment (HonkingGoose) [inline .github/ISSUE_TEMPLATE/tech-debt.yml:1]:
```suggestion
name: ⚙ Tech Debt
```

**Comment by HonkingGoose** [comment]
> I observed a few extra changes on top of the migration from templates to forms, but I generally agree with those too, so this looks great to me.

You're correct, I changed some style, capitalization and grammar things as well in this PR.

> I didn't know about this feature, it's very nice!

I like it as well! It makes opening a bug report or filing a feature request a lot easier. 😄

**Review by deivid-rodriguez — APPROVED** [review]

**Pull request merged**